src/storage.c

```diff
@@ -102,6 +103,34 @@ static int save_settings(int fd, struct l_settings *settings)
 	return err;
 }
 
+static int check_valid_groups(char **all_groups, int index)
+{
+	int repeated_check;
+	regex_t preg;
+	int i;
+	int err;
+
+	regcomp(&preg, "^DataItem_[0-9]+$", REG_EXTENDED);
+
+	if (!regexec(&preg, all_groups[index], 0, (regmatch_t *)NULL, 0) == 0) {
+		l_error("Invalid DataItem group: %s", all_groups[index]);
+		err = -EINVAL;
```

Comment:
You can immediately return the error here to avoid entering the for loop.

Reply:
Done!